src/components/OnboardingModal/OnboardingModal.tsx

```diff
@@ -129,14 +151,39 @@ export default function OnboardingModal() {
   const handleAcceptTOS = () => {
     acceptTOS(undefined, {
       async onSuccess() {
-        setActiveStep((x) => x + 1);
+        goNext();
       },
     });
   };
-  const handleCompleteOnboarding = () => {
-    completeOnboarding();
+  const handleCompleteStep = (step: OnboardingStep) => {
+    completeStep(
+      { step },
+      {
+        onSuccess: (result) => {
+          if (result.onboardingSteps.length > 0) {
+            goNext();
+            return;
+          }
+
+          if (showReferral && user)
+            mutate({ ...user, userReferralCode: userReferral.code, source: userReferral.source });
+        },
+      }
+    );
+  };
+  const handleCompleteBuzzStep = () => {
+    setReferralError('');
+
+    const result = referralSchema.safeParse(userReferral);
+    if (!result.success)
+      return setReferralError(result.error.format().code?._errors[0] ?? 'Invalid value');
+    // if (referrerLoading) return;
```

Comment:
Might make sense to keep in case user tries to click right after inputing the code?